code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -19,18 +19,16 @@
 
 	if(href_list["PRG_openfile"])
 		. = 1
+		var/passcheck
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
-		var/datum/computer_file/file = HDD.find_file_by_name(href_list["PRG_openfile"])
-		var/checkpass = ""
-		var/currentpass = file.password
-		if(!currentpass)
+		var/datum/computer_file/F = HDD.find_file_by_name(href_list["PRG_openfile"])
+		var/iscorrect = F.can_acess_file(passcheck)
```

Comment:
This needs to be after line 27. We don't know if F exists yet.

Reply:
The var `iscorrect` is unnecessary. Just call the proc in an if: `if (F.can_access_file(passcheck))`